src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -102,42 +101,21 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
                     parameters.VersionName,
                     parameters.ApplyTemplateSettings,
                     parameters.RootTocPath);
-                IHostServiceCreator hostServiceCreator = null;
-                PhaseProcessor phaseProcessor = null;
-                if (ShouldTraceIncrementalInfo)
-                {
-                    context.IncrementalBuildContext = IncrementalBuildContext.Create(parameters, CurrentBuildInfo, LastBuildInfo, IntermediateFolder);
-                    hostServiceCreator = new HostServiceCreatorWithIncremental(context);
-                    phaseProcessor = new PhaseProcessor
-                    {
-                        Handlers =
-                        {
-                            new PrebuildBuildPhaseHandlerWithIncremental(context),
-                            new PostbuildPhaseHandlerWithIncremental(context),
-                        }
-                    };
-                }
-                else
-                {
-                    hostServiceCreator = new HostServiceCreator(context);
-                    phaseProcessor = new PhaseProcessor
-                    {
-                        Handlers =
-                        {
-                            new PrebuildBuildPhaseHandler(context),
-                            new PostbuildPhaseHandler(context),
-                        }
-                    };
-                }
 
                 Logger.LogVerbose("Start building document...");
 
                 // Start building document...
                 List<HostService> hostServices = null;
+                IHostServiceCreator hostServiceCreator = null;
+                PhaseProcessor phaseProcessor = null;
                 try
                 {
                     using (var templateProcessor = parameters.TemplateManager?.GetTemplateProcessor(context, parameters.MaxParallelism) ?? TemplateProcessor.DefaultProcessor)
                     {
+                        using (new LoggerPhaseScope("Prepare", true))
+                        {
+                            Prepare(parameters, context, templateProcessor, out hostServiceCreator, out phaseProcessor);
```

Comment:
>Prepare [](start = 28, length = 7)

what if Prepare fails, e.g. incremental files get deleted, fall back to clean build?